ribs/visualize.py

```diff
@@ -496,102 +515,116 @@ def parallel_axes_plot(archive,
     vmin = np.min(df['objective']) if vmin is None else vmin
     vmax = np.max(df['objective']) if vmax is None else vmax
     norm = matplotlib.colors.Normalize(vmin=vmin, vmax=vmax, clip=True)
     objectives = df['objective'].to_numpy()
     ys = df[cols].to_numpy()
     y_ranges = upper_bounds - lower_bounds
 
-    # transform all data to be in the first axis coordinates
-    zs = np.zeros_like(ys)
-    zs[:, 0] = ys[:, 0]
-    zs[:, 1:] = (ys[:, 1:] - lower_bounds[1:]) / \
-                y_ranges[1:] * y_ranges[0] + lower_bounds[0]
+    # Transform all data to be in the first axis coordinates.
+    normalized_ys = np.zeros_like(ys)
+    normalized_ys[:, 0] = ys[:, 0]
+    normalized_ys[:, 1:] = ((ys[:, 1:] - lower_bounds[1:]) /
+                y_ranges[1:] * y_ranges[0] + lower_bounds[0])
 
-    #copy the axis for the other bcs
-    axes = [host] + [host.twinx() for i in range(ys.shape[1] - 1)]
+    # Copy the axis for the other bcs.
+    axes = [host_ax] + [host_ax.twinx() for i in range(len(cols) - 1)]
     for i, axis in enumerate(axes):
         axis.set_ylim(lower_bounds[i], upper_bounds[i])
         axis.spines['top'].set_visible(False)
         axis.spines['bottom'].set_visible(False)
-        if axis != host:
+        if axis != host_ax:
             axis.spines['left'].set_visible(False)
             axis.yaxis.set_ticks_position('right')
-            axis.spines["right"].set_position(("axes", i / (ys.shape[1] - 1)))
-
-    host.set_xlim(0, ys.shape[1] - 1)
-    host.set_xticks(range(ys.shape[1]))
-    labels = cols if axis_labels is None else axis_labels
-    host.set_xticklabels(labels)
-    host.tick_params(axis='x', which='major', pad=7)
-    host.spines['right'].set_visible(False)
-    host.xaxis.tick_top()
-
-    for j in range(len(archive_data)):
-        #draw straight lines between the axes in the appropriate color
-        color = matplotlib.colors.rgb2hex(cmap(norm(objectives[j])))
-        host.plot(range(ys.shape[1]),
-                  zs[j, :],
+            axis.spines["right"].set_position(("axes", i / (len(cols) - 1)))
+
+    host_ax.set_xlim(0, len(cols) - 1)
+    host_ax.set_xticks(range(len(cols)))
+    host_ax.set_xticklabels(axis_labels)
+    host_ax.tick_params(axis='x', which='major', pad=7)
+    host_ax.spines['right'].set_visible(False)
+    host_ax.xaxis.tick_top()
+
+    for archive_entry, objective in zip(normalized_ys, objectives):
+        # Draw straight lines between the axes in the appropriate color.
+        color = cmap(norm(objective))
+        host_ax.plot(range(len(cols)),
+                  archive_entry,
                   c=color,
                   alpha=alpha,
                   linewidth=linewidth)
 
     # Create a colorbar.
     mappable = ScalarMappable(cmap=cmap)
     mappable.set_clim(vmin, vmax)
-    host.figure.colorbar(mappable, pad=0.05, orientation='horizontal')
+    host_ax.figure.colorbar(mappable, pad=cbar_pad, orientation=cbar_orientation)
```

Comment:
Put an assertion at the beginning of the function that checks `cbar_orientation` is `horizontal` or `vertical`. It would be bad if the user waited for the lines to plot, only to find that they failed due to a parameter error.